src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs

```diff
@@ -90,6 +92,35 @@ await foreach (var tuple in _kubernetesService.WatchAsync<T>(cancellationToken:
         });
     }
 
+    /// <summary>
+    /// This is workaround for https://github.com/dotnet/aspire/issues/1364 where there is a timing issue where an existing connection to Traefik
+    /// will return a 404 for potentially several minutes if that connection is made before it is finished loading its configuration. The below code
+    /// will hit the proxy endpoint setup in that configuration until we do not see the known 404 response from Traefik.
+    /// </summary>
+    private void RunServiceWatchTask()
+    {
+        var maxRetryDuration = TimeSpan.FromSeconds(5);
+
+        _ = Task.Run(async () =>
+        {
+            List<Task> tasks = new List<Task>();
+
+            try
+            {
+                await foreach ((WatchEventType eventType, Service service) in _kubernetesService.WatchAsync<Service>(cancellationToken: _cancellationToken))
+                {
+                    tasks.Add(PollUntilServiceEndpointValid(eventType, service, maxRetryDuration));
+                }
+
+                await Task.WhenAll(tasks).ConfigureAwait(false);
```

Comment:
kubernetes watch generally don't terminate. Are we actually hitting this code path? Given we are using channel, we can just launch the task in await foreach above and let it write to channel whenever it finishes. cancellation token would make sure it would be cancelled

Reply:
@danegsta ?

Reply:
We wouldn't log an error in the catch block below if we don't await on the tasks; functionally it's not necessary, but it arguably helps with logging if anything goes wrong.

Reply:
Interesting. What I am talking is all theory so keep this line if it is causing some behavior difference.